recipes/opencv/4.x/conanfile.py

```diff
@@ -306,7 +312,13 @@ def add_components(components):
                         if conan_component == "opencv_core":
                             self.cpp_info.components[conan_component].libdirs.append("lib")
                             self.cpp_info.components[conan_component].libs += tools.collect_libs(self)
- 
+
+                if self.settings.os == "iOS":
+                    if not self.options.shared:
+                        if conan_component == "opencv_core":
+                            libs = list(filter(lambda x: not x.startswith("opencv"), tools.collect_libs(self)))
```

Comment:
iOS static builds generate multiple libs starting with `opencv_core`

🤔 ... 😕

Reply:
@prince-chrismc Not sure if I get it what you mean... Is it about the code duplications? Or do you mean to add `requires` for these libs? For the latter case, I'm not sure if everything which is generated here is available as recipe... But for iOS, I'm currently trying to extract `quirc` to an requires, which would make this piece of code obsolete.

Reply:
Please, add the expect library name instead.

Reply:
That's the problem here, I cannot say what's the expected library name. I found 2 (cpufeatures for Android and tegra_hal for Android iOS and maybe others) but there are possibly others, with not so obvious dependencies to options and platform. What I try here is to collect this libraries dynamically. As it is already done for Android, BTW.